File: backend/src/main/java/com/tripfriend/domain/trip/schedule/dto/TripScheduleReqDto.java
package com.tripfriend.domain.trip.schedule.dto;

import com.tripfriend.domain.trip.information.dto.TripInformationReqDto;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class TripScheduleReqDto {
    private Long memberId;
    private String title;
    private String cityName;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;
    private List<TripInformationReqDto> tripInformations;
}

File: backend/src/main/java/com/tripfriend/domain/trip/schedule/dto/TripScheduleReqDto.kt
package com.tripfriend.domain.trip.schedule.dto

import com.tripfriend.domain.trip.information.dto.TripInformationReqDto
import com.tripfriend.domain.trip.schedule.entity.TripSchedule
import java.time.LocalDate

data class TripScheduleReqDto(
    val memberId: Long?,
    val title: String?,
    val cityName: String? = null,
    val description: String?,
    val startDate: LocalDate?,
    val endDate: LocalDate?,
    val tripInformations: MutableList<TripInformationReqDto> = mutableListOf()
) {
    constructor(trip: TripSchedule) : this(
        memberId = trip.member?.id,
        title = trip.title,
        cityName = trip.tripInformations[0].place?.cityName,
        description = trip.description,
        startDate = trip.startDate,
        endDate = trip.endDate,
        tripInformations = trip.tripInformations.map { TripInformationReqDto() }.toMutableList()
    )
}
